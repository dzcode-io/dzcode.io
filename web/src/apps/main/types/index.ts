export interface SidebarTreeItem {
  id: string;
  content: string;
  link?: string;
  children?: SidebarTreeItem[];
}
