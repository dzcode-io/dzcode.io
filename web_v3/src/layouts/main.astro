<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <!-- @TODO-ZM: add head tags -->
  </head>
  <body>
    <slot />
  </body>
</html>
