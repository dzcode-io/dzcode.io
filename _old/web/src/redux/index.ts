import { combineReducers, compose, configureStore } from "@reduxjs/toolkit";
import * as slices from "src/redux/reducers";

// -----------------------------------------------------------------------------
type Slices = typeof slices;
export type SlicesKey = keyof typeof slices;
type Reducers = { [K in keyof Slices]: Slices[K]["reducer"] };
type Actions = { [K in keyof Slices]: Slices[K]["actions"] };
export type ActionType = {
  [K in keyof Actions]: `${K}/${keyof Actions[K] & string}`;
}[SlicesKey];
export type State = ReturnType<typeof rootReducer>;
// -----------------------------------------------------------------------------

const reducers = (Object.keys(slices) as SlicesKey[]).reduce(
  (pV, sliceKey) => ({ ...pV, [sliceKey]: slices[sliceKey].reducer }),
  {} as Reducers,
);
const rootReducer = combineReducers(reducers);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const createStore = () => configureStore({ reducer: rootReducer, enhancers: composeEnhancers });

let store = createStore();

type GetStoreParam = {
  cacheStore?: boolean;
};

export const getStore = ({ cacheStore = false }: GetStoreParam = {}) => {
  if (!cacheStore) {
    store = createStore();
  }
  return store;
};

// @TODO-ZM: cache this, and add a subscriber that refreshes the cache whenever new actions get dispatched
export const getState = store.getState;

export const actions = (Object.keys(slices) as SlicesKey[]).reduce(
  (pV, sliceKey) => ({
    ...pV,
    [sliceKey]: Object.keys(slices[sliceKey].actions).reduce(
      (contextualActions, actionName) => ({
        ...contextualActions,
        [actionName]: (...args: any[]) =>
          store.dispatch((slices[sliceKey].actions as any)[actionName](...args)),
      }),
      {},
    ),
  }),
  {} as Actions,
);
