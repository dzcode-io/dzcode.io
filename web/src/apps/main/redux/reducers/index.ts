import { articles } from "./articles";
import { articlesPage } from "./articles-page";
import { combineReducers } from "redux";
import { contributePage } from "./contribute-page";
import { documentation } from "./documentation";
import { faqPage } from "./faq-page";
import { footerComponent } from "./footer-component";
import { landingPage } from "./landing-page";
import { learnPage } from "./learn-page";
import { navbarComponent } from "./navbar-component";
import { projectsPage } from "./projects-page";
import { settings } from "./settings";
import { teamPage } from "./team-page";

export const mainReducer = combineReducers({
  documentation,
  learnPage,
  articles,
  articlesPage,
  projectsPage,
  landingPage,
  settings,
  faqPage,
  navbarComponent,
  footerComponent,
  contributePage,
  teamPage,
});
