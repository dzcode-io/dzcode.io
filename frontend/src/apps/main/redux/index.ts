import { createStore, compose } from "redux";
import { applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { mainReducer } from "./reducers";

const composeEnhancers =
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const mainStore = createStore(
  mainReducer,
  composeEnhancers(applyMiddleware(thunk)),
);
