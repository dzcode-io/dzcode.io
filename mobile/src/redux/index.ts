import { applyMiddleware, compose, createStore } from "redux";
import thunk, { ThunkAction, ThunkDispatch } from "redux-thunk";

import { mainReducer } from "./reducers";

/**
 * the main redux state, with all the reducers
 */
export const mainStore = createStore(mainReducer, compose(applyMiddleware(thunk)));

/**
 * Creates a new redux state each time this function is called, this is used only for unit tests, to ensure that we have fresh state on each individual test
 */
export const createMainStore = () => {
  return createStore(mainReducer, compose(applyMiddleware(thunk)));
};

export type StateInterface = ReturnType<typeof mainStore.getState>;

export type ActionType =
  | "UPDATE_SETTINGS"
  | "UPDATE_GENERAL"
  | "UPDATE_CONTRIBUTE_SCREEN"
  | "UPDATE_ARTICLES_SCREEN"
  | "UPDATE_LEARN_SCREEN"
  | "UPDATE_PROJECTS_SCREEN";

export interface Action<T> {
  type: ActionType;
  payload: Partial<T>;
}

export type ThunkResult<A = Record<string, unknown>, E = Record<string, unknown>> = ThunkAction<
  void,
  StateInterface,
  E,
  Action<A>
>;

export type Dispatch<A> = ThunkDispatch<StateInterface, Record<string, unknown>, Action<A>>;
