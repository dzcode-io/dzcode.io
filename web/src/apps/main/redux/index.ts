import { applyMiddleware, compose, createStore } from "redux";
import thunk, { ThunkAction, ThunkDispatch } from "redux-thunk";

import { mainReducer } from "./reducers";

const composeEnhancers =
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/**
 * the main redux state, with all the reducers
 */
export const mainStore = createStore(mainReducer, composeEnhancers(applyMiddleware(thunk)));

/**
 * Creates a new redux state each time this function is called, this is used only for unit tests, to ensure that we have fresh state on each individual test
 */
export const createMainStore = () => {
  return createStore(mainReducer, composeEnhancers(applyMiddleware(thunk)));
};

export type StateInterface = ReturnType<typeof mainStore.getState>;

export type ActionType =
  | "UPDATE_DOCUMENTATION"
  | "UPDATE_LEARN_PAGE"
  | "UPDATE_ARTICLES"
  | "UPDATE_ARTICLES_PAGE"
  | "UPDATE_PROJECTS"
  | "UPDATE_PROJECTS_PAGE"
  | "UPDATE_LANDING_PAGE"
  | "UPDATE_SETTINGS"
  | "UPDATE_CONTRIBUTE_PAGE";

export interface Action<T> {
  type: ActionType;
  payload: Partial<T>;
}

export type ThunkResult<A = Record<string, unknown>, E = Record<string, unknown>> = ThunkAction<
  void,
  StateInterface,
  E,
  Action<A>
>;

export type Dispatch<A> = ThunkDispatch<StateInterface, Record<string, unknown>, Action<A>>;
