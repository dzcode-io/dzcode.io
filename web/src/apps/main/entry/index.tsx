import * as Sentry from "@sentry/browser";
import { App } from "./app";
import { Integrations } from "@sentry/tracing";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { getEnv } from "src/common/utils";
import { mainStore } from "src/apps/main/redux";
import { render } from "react-dom";

Sentry.init({
  dsn: "https://b42ee25f06c4416dac83de4ab8772e26@o953637.ingest.sentry.io/5902610",
  integrations: [new Integrations.BrowserTracing()],
  tracesSampleRate: 1.0,
  environment: getEnv(),
});

render(
  <Provider store={mainStore}>
    <Router>
      <App />
    </Router>
  </Provider>,
  document.getElementById("app-container"),
);
