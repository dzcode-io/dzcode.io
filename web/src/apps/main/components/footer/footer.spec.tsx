import { Footer } from ".";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { mainStore } from "src/apps/main/redux";
import { render } from "@testing-library/react";

describe("components/footer/footer.spec.tsx", () => {
  test("should render properly", () => {
    const { container } = render(
      <Provider store={mainStore}>
        <Router>
          <Footer />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
