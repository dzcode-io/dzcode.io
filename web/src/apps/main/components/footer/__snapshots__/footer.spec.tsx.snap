// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/footer/footer.spec.tsx should render properly 1`] = `
<div>
  <footer
    class="makeStyles-root-1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-9"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
          >
            <h6
              class="MuiTypography-root makeStyles-categoryTitle-3 MuiTypography-h6"
            >
              Helpful Links
            </h6>
            <a
              href="/"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Home
              </h6>
            </a>
            <a
              href="/Learn"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Learn
              </h6>
            </a>
            <a
              href="/Projects"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Projects
              </h6>
            </a>
            <a
              href="/Articles"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Articles
              </h6>
            </a>
            <a
              href="/FAQ"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                FAQ
              </h6>
            </a>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
          >
            <h6
              class="MuiTypography-root makeStyles-categoryTitle-3 MuiTypography-h6"
            >
              Mobile
            </h6>
            <a
              href="/http://localhost:19002"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Android
              </h6>
            </a>
            <a
              href="/http://localhost:19002"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                iOS
              </h6>
            </a>
            <a
              href="/http://localhost:19002"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Expo
              </h6>
            </a>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
          >
            <h6
              class="MuiTypography-root makeStyles-categoryTitle-3 MuiTypography-h6"
            >
              Social Media
            </h6>
            <a
              href="https://www.github.com/dzcode-io"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Github
              </h6>
            </a>
            <a
              href="https://join.slack.com/t/dzcode/shared_invite/zt-ek9kscb7-m8z_~cBjX79l~uchuABPFQ"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Slack
              </h6>
            </a>
            <a
              href="https://www.facebook.com/dzcode.io"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Facebook
              </h6>
            </a>
            <a
              href="https://www.instagram.com/dzcode.io"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Instagram
              </h6>
            </a>
            <a
              href="https://www.youtube.com/channel/UC_tLjuQaYotzERtaAo8Y4SQ"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                YouTube
              </h6>
            </a>
            <a
              href="https://twitter.com/dzcode_io"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                Twitter
              </h6>
            </a>
            <a
              href="https://www.linkedin.com/groups/8924363"
            >
              <h6
                class="MuiTypography-root makeStyles-linkText-2 MuiTypography-subtitle2"
              >
                LinkedIn
              </h6>
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-contactDetails-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3"
        >
          <h6
            class="MuiTypography-root makeStyles-categoryTitle-3 MuiTypography-h6"
          >
            Contact Information
          </h6>
          <a
            href="tel:+21367-626-1157"
          >
            <p
              class="MuiTypography-root makeStyles-linkText-2 MuiTypography-body1"
            >
              +213 06-76-26-11-57
            </p>
          </a>
          <a
            href="mailto:contact@dzcode.io"
          >
            <p
              class="MuiTypography-root makeStyles-linkText-2 MuiTypography-body1"
            >
              contact@dzcode.io
            </p>
          </a>
          <p
            class="MuiTypography-root makeStyles-linkText-2 MuiTypography-body1"
          >
            Copyright Â© 
            2021 
            <a
              href="https://twitter.com/dzcode_io"
            >
              @dzCode_io
            </a>
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
